linesearch.go

```diff
@@ -62,6 +62,11 @@ func (ls *Linesearch) Iterate(loc *Location, xNext []float64) (EvaluationType, I
 		return NoEvaluation, ls.iterType, nil
 	}
 	if ls.iterType == MajorIteration {
+		// Compare the starting points for the previous and next iteration to
```

Comment:
Okay, I apparently can't inline comment where code didn't change. 

I think the problem is when x == x + stepsize * dir due to floating point errors. The error is that the linesearch can't make progress even though it thinks it can. This is then a "linesearch failed to converge" error. I don't think it's necessarily an error if the location at two major iterations is the same, for example the NextDirectioner wants to linesearch in a different direction starting from the same location for whatever reason.